### PR TITLE
tsp, partial-update, replace static initializer declaration

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
partly solve https://github.com/Azure/autorest.java/issues/2468

assumption is this block is always treat as `@Generated` https://github.com/Azure/autorest.java/issues/2468#issuecomment-1897654248, and (since it does not have a name) always corresponding.

code https://github.com/Azure/autorest.java/pull/2593/commits/8417747019b76988f42023216e1b9baed83fab47

### CODE REVIEW

File: partial-update/src/main/java/com/azure/autorest/partialupdate/util/PartialUpdateHandler.java [hunk 1]
```diff
@@ -474,6 +490,11 @@ private static boolean isMembersCorresponding(BodyDeclaration<?> member1, BodyDe
         if (member1.isCallableDeclaration() && member2.isCallableDeclaration()) {
             // compare signature
             return member1.asCallableDeclaration().getSignature().equals(member2.asCallableDeclaration().getSignature());
+        } else if (member1 instanceof InitializerDeclaration && member2 instanceof InitializerDeclaration
+                && ((InitializerDeclaration) member1).isStatic() && ((InitializerDeclaration) member2).isStatic()) {
```
Comment (haolingdong-msft) [inline partial-update/src/main/java/com/azure/autorest/partialupdate/util/PartialUpdateHandler.java:494]: so assumption is generated static block is already same with existing static block? would add this into comment?

Comment (weidongxu-microsoft) [reply partial-update/src/main/java/com/azure/autorest/partialupdate/util/PartialUpdateHandler.java:494]: No such assumption.

It is "always replace" (same logic, if method on both side is `@Generated`).

**Review by haolingdong-msft — APPROVED** [review]

File: partial-update/src/test/java/com/azure/autorest/partialupdate/util/PartialUpdateHandlerTest.java [hunk 2]
```diff
@@ -408,4 +412,64 @@ public void testModuleInfoFileToTestMergeExistingFileAndGeneratedFileToTestIgnor
...
+        assertEquals(1, compilationUnit.getTypes().size());
+        List<InitializerDeclaration> staticInitializerDeclaration = compilationUnit.getTypes().get(0).getMembers().stream()
+                .filter(m -> m instanceof InitializerDeclaration)
+                .map(m -> (InitializerDeclaration) m)
+                .filter(InitializerDeclaration::isStatic)
+                .collect(Collectors.toList());
+        // verify 1 block
+        Assertions.assertEquals(1, staticInitializerDeclaration.size());
+        // verify the block is replaced
+        Assertions.assertTrue(staticInitializerDeclaration.get(0).toString().contains("// token for test"));
```
Comment (weidongxu-microsoft) [inline partial-update/src/test/java/com/azure/autorest/partialupdate/util/PartialUpdateHandlerTest.java:432]: test for this "replace"

File: partial-update/src/main/java/com/azure/autorest/partialupdate/util/PartialUpdateHandler.java (hunk 1 above)
Comment (haolingdong-msft) [reply partial-update/src/main/java/com/azure/autorest/partialupdate/util/PartialUpdateHandler.java:494]: Sorry for using the wrong word 'same',  'corresponding' is better. both `@Generated` and method/code block is corresponding, then we will replace.

Comment (weidongxu-microsoft) [reply partial-update/src/main/java/com/azure/autorest/partialupdate/util/PartialUpdateHandler.java:499]: Added the comment to Javadoc of the method as well.

**Review by haolingdong-msft — APPROVED** [review]

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**